File: examples/src/SmartArts/AddNodes.cs
 using System.IO
import aspose.slides as slides
using Aspose.slides.Examples.CSharp
namespace Aspose.slides.Examples.CSharp.SmartArts

{
    public class AddNodes
    {
        public static void Run()
        {
            #ExStart:AddNodes
            # The path to the documents directory.
            dataDir = RunExamples.GetDataDir_SmartArts()

            # Load the desired the presentation# Load the desired the presentation
            Presentation pres = new Presentation(dataDir+ "AddNodes.pptx")

            # Traverse through every shape inside first slide
            foreach (IShape shape in pres.slides[0].shapes)
            {

                # Check if shape is of SmartArt type
                if (shape is Aspose.slides.SmartArt.SmartArt)
                {

                    # Typecast shape to SmartArt
                    Aspose.slides.SmartArt.SmartArt smart = (Aspose.slides.SmartArt.SmartArt)shape

                    # Adding a new SmartArt Node
                    Aspose.slides.SmartArt.SmartArtNode TemNode = (Aspose.slides.SmartArt.SmartArtNode)smart.AllNodes.AddNode()

                    # Adding text
                    TemNode.text_frame.text = "Test"

                    # Adding new child node in parent node. It  will be added in the end of collection
                    Aspose.slides.SmartArt.SmartArtNode newNode = (Aspose.slides.SmartArt.SmartArtNode)TemNode.ChildNodes.AddNode()

                    # Adding text
                    newNode.text_frame.text = "New Node Added"

                }
            }

            # Saving Presentation
            pres.save(dataDir + "AddSmartArtNode_out.pptx", slides.export.SaveFormat.PPTX)
            #ExEnd:AddNodes
        }
    }
}

File: examples/src/smart_arts/add_nodes.py
import aspose.slides as slides


def add_nodes(global_opts):
    # Load the desired the presentation# Load the desired the presentation
    with slides.Presentation(global_opts.data_dir + "smart_art_access.pptx") as pres:
        # Traverse through every shape inside first slide
        for shape in pres.slides[0].shapes:
            # Check if shape is of SmartArt type
            if type(shape) is slides.smartart.SmartArt:
                # Typecast shape to SmartArt
                smart = shape

                # Adding a new SmartArt Node
                temp_node = smart.all_nodes.add_node()

                # Adding text
                temp_node.text_frame.text = "Test"

                # Adding new child node in parent node. It  will be added in the end of collection
                new_node = temp_node.child_nodes.add_node()

                # Adding text
                new_node.text_frame.text = "New Node Added"

        # Saving Presentation
        pres.save(global_opts.out_dir + "smart_art_add_node_out.pptx", slides.export.SaveFormat.PPTX)
